unit_tests/test_redis_aggregator.py

```diff
@@ -0,0 +1,238 @@
+    def test_execute_batch_commands_api_time_out_raise_exception(aggregator):
+        with patch("gsy_e_sdk.redis_aggregator.wait_until_timeout_blocking",
+                   side_effect=AssertionError):
+            aggregator.device_uuid_list = [TEST_DEVICE_UUID_1, TEST_DEVICE_UUID_2]
+            aggregator.aggregator_uuid = TEST_AGGREGATOR_UUID
+            with pytest.raises(RedisAggregatorAPIException):
+                aggregator.execute_batch_commands()
+
+    @staticmethod
+    @pytest.mark.usefixtures("mock_transaction_id_and_timeout_blocking")
+    @pytest.mark.parametrize("is_blocking_, trans_id_resp_buffer, expected_ret_val",
+                             [(True, TEST_TRANSACTION_ID_RESPONSE_BUFFER, TEST_RESPONSE),
+                              (False, TEST_TRANSACTION_ID_RESPONSE_BUFFER, None),
+                              (True, {}, None)])
+    def test_execute_batch_commands_returns_expected(aggregator, is_blocking_,
+                                                     trans_id_resp_buffer, expected_ret_val):
+        aggregator.device_uuid_list = [TEST_DEVICE_UUID_1, TEST_DEVICE_UUID_2]
+        aggregator.aggregator_uuid = TEST_AGGREGATOR_UUID
+        aggregator._transaction_id_response_buffer = trans_id_resp_buffer
+
+        assert aggregator.execute_batch_commands(is_blocking=is_blocking_) is expected_ret_val
+
+
+@pytest.mark.usefixtures("mock_grid_fee_calculation")
+class TestRedisAggregatorCalculateGridFee:
+    """Test cases for RedisAggregator's calculate_grid_fee method."""
+
+    @staticmethod
+    def test_calculate_grid_fee_called_with_args(aggregator):
+        aggregator.calculate_grid_fee("start_market",
+                                      "target_market",
+                                      "current_market_fee")
+        aggregator.grid_fee_calculation.calculate_grid_fee.assert_called_with(
+            "start_market",
+            "target_market",
+            "current_market_fee")
+
+    @staticmethod
+    def test_calculate_grid_fee_returns_expected(aggregator):
+        aggregator.grid_fee_calculation.calculate_grid_fee.return_value = "TEST_OK"
+        ret_val = aggregator.calculate_grid_fee("start_market",
+                                                "target_market",
+                                                "current_market_fee")
+        assert ret_val == "TEST_OK"
+
+
+def test_get_uuid_from_area_name_returns_expected(aggregator):
+    aggregator.area_name_uuid_mapping = TEST_AREA_MAPPING
+    area_name = TEST_AREA_NAME
+    ret_val = TEST_AREA_MAPPING[TEST_AREA_NAME][0]
+
+    patch("gsy_e_sdk.redis_aggregator.get_uuid_from_area_name_in_tree_dict",
+          return_value=ret_val)
+    assert aggregator.get_uuid_from_area_name(area_name) == ret_val
+
+
+def test_add_to_batch_commands_returns_client_command_buffer(aggregator):
+    buffer_instance = aggregator.add_to_batch_commands
+    assert buffer_instance == aggregator._client_command_buffer
```

Comment:
Once again, my attention was caught by the private `_client_command_buffer`. Since it's private, I think that we probably shouldn't know about it. Here we want to know that `self.add_to_batch_commands` will return the same instance of `ClientCommand` that was instantiated in the `__init__()` method.

Reply:
I addressed your comment by implementing something similar to [this](https://github.com/gridsingularity/gsy-e-sdk/pull/197#discussion_r773360128).